.github/workflows/branch-deploy.yml

```diff
@@ -89,7 +90,9 @@ jobs:
 
       # this step is custom to my site and adds a 'commit' version to the site
       - name: write build version
-        run: echo ${GITHUB_SHA} > ./public/version.txt
+        env:
+          BUILD_SHA: ${{ needs.trigger.outputs.sha }}
+        run: echo ${BUILD_SHA} > ./public/version.txt
```

Comment:
Unless we specifically use the `trigger.outputs.sha` value from the branch-deploy Action, this old `GITHUB_SHA` would be the commit on `main` as that is the context of this workflow since it is from an `issue_comment`